### ISSUE TITLE
Feature Extraction problem with ucf-101

### ISSUE THREAD

**Issue by MarcoSaku** [facebook/C3D#53]
Hi! I followed the guide and I trained on a subset of ucf-101 dataset. I used 3 classes (I changed the num_output of last fc to 3) and it works. I have a strange problem when I use the script /build/tools/extract_image_features.bin. If I use it in demo it works (in c3d_feature_extraction folder), but if I use it with my training it doesn't work. 
I use the .prototxt in c3d_train_ucf101 and the snapshot created in the training. 
The script is this one:
```GLOG_logtosterr=1 ../../build/tools/extract_image_features.bin ./proto_files/conv3d_ucf101_test_nobias.prototxt ./snapshots/conv3d_ucf101_3classes_titan_nobias_iter_450 0 1 1 ./proto_files/output_list_prefix.txt prob```

I have the following error:
I1217 18:49:40.410287 23534 net.cpp:76] Creating Layer fc8
I1217 18:49:40.410292 23534 net.cpp:86] fc8 <- fc7
I1217 18:49:40.410297 23534 net.cpp:112] fc8 -> fc8
I1217 18:49:40.410418 23534 net.cpp:127] Top shape: 1 3 1 1 1 (3)
I1217 18:49:40.410436 23534 net.cpp:154] fc8 needs backward computation.
I1217 18:49:40.410444 23534 net.cpp:76] Creating Layer prob
I1217 18:49:40.410449 23534 net.cpp:86] prob <- fc8
I1217 18:49:40.410459 23534 net.cpp:112] prob -> prob
I1217 18:49:40.410466 23534 net.cpp:127] Top shape: 1 3 1 1 1 (3)
I1217 18:49:40.410471 23534 net.cpp:154] prob needs backward computation.
I1217 18:49:40.410477 23534 net.cpp:76] Creating Layer accuracy
I1217 18:49:40.410482 23534 net.cpp:86] accuracy <- prob
I1217 18:49:40.410487 23534 net.cpp:86] accuracy <- label
I1217 18:49:40.410500 23534 net.cpp:112] accuracy -> accuracy
I1217 18:49:40.410524 23534 net.cpp:127] Top shape: 1 2 1 1 1 (2)
I1217 18:49:40.410531 23534 net.cpp:154] accuracy needs backward computation.
I1217 18:49:40.410536 23534 net.cpp:165] This network produces output accuracy
I1217 18:49:40.410553 23534 net.cpp:183] Collecting Learning Rate and Weight Decay.
I1217 18:49:40.410573 23534 net.cpp:176] Network initialization done.
I1217 18:49:40.410578 23534 net.cpp:177] Memory required for Data 103796772
 in the network ./proto_files/conv3d_ucf101_test_nobias.prototxt failed: feature_extraction_net->has_blob(string(argv[i])) Unknown feature blob name prob
*** Check failure stack trace: ***
Aborted (core dumped)

As you can see the layer prob is present. Maybe to use this script I need to do something when I train the model?
Thanks

**Comment by dutran** [comment]
@MarcoSaku This says that you prototxt file has not layer named "prob" but you want to extract feature with name prob. Please correct your "./proto_files/conv3d_ucf101_test_nobias.prototxt"

**Comment by MarcoSaku** [comment]
This is my prototxt. As you can see there is a layer "prob". I tried also with other layers and it doesn't work

./proto_files/conv3d_ucf101_test_nobias.prototxt

name: "deep_c3d_ucf101"
layers {
  name: "data"
  type: VIDEO_DATA
  top: "data"
  top: "label"
  image_data_param {
    source: "./lst_files/my_test_3classes.lst"
    use_image: true
    mean_file: "ucf101_train_mean.binaryproto"
    batch_size: 4
    crop_size: 112
    mirror: false
    show_data: 0
    new_height: 128
    new_width: 171
    new_length: 16
    shuffle: false
  }
}
# ----------- 1st layer group ---------------
layers {
  name: "conv1a"
  type: CONVOLUTION3D
  bottom: "data"
  top: "conv1a"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 64
    kernel_size: 3
    kernel_depth: 3
    pad: 1
    temporal_pad: 1
    stride: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu1a"
  type: RELU
  bottom: "conv1a"
  top: "conv1a"
}
layers {
  name: "pool1"
  type: POOLING3D
  bottom: "conv1a"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 1
    stride: 2
    temporal_stride: 1
  }
}
# ------------- 2nd layer group --------------
layers {
  name: "conv2a"
  type: CONVOLUTION3D
  bottom: "pool1"
  top: "conv2a"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 128
    kernel_size: 3
    kernel_depth: 3
    pad: 1
    temporal_pad: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu2a"
  type: RELU
  bottom: "conv2a"
  top: "conv2a"
}
layers {
  name: "pool2"
  type: POOLING3D
  bottom: "conv2a"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 2
    stride: 2
    temporal_stride: 2
  }
}
# ----------------- 3rd layer group --------------
layers {
  name: "conv3a"
  type: CONVOLUTION3D
  bottom: "pool2"
  top: "conv3a"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 256
    kernel_size: 3
    kernel_depth: 3
    pad: 1
    temporal_pad: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu3a"
  type: RELU
  bottom: "conv3a"
  top: "conv3a"
}
layers {
  name: "pool3"
  type: POOLING3D
  bottom: "conv3a"
  top: "pool3"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 2
    stride: 2
    temporal_stride: 2
  }
}

# --------- 4th layer group
layers {
  name: "conv4a"
  type: CONVOLUTION3D
  bottom: "pool3"
  top: "conv4a"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 256
    kernel_size: 3
    kernel_depth: 3
    pad: 1
    temporal_pad: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu4a"
  type: RELU
  bottom: "conv4a"
  top: "conv4a"
}
layers {
  name: "pool4"
  type: POOLING3D
  bottom: "conv4a"
  top: "pool4"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 2
    stride: 2
    temporal_stride: 2
  }
}

# --------------- 5th layer group --------
layers {
  name: "conv5a"
  type: CONVOLUTION3D
  bottom: "pool4"
  top: "conv5a"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  convolution_param {
    num_output: 256
    kernel_size: 3
    kernel_depth: 3
    pad: 1
    temporal_pad: 1
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu5a"
  type: RELU
  bottom: "conv5a"
  top: "conv5a"
}
layers {
  name: "pool5"
  type: POOLING3D
  bottom: "conv5a"
  top: "pool5"
  pooling_param {
    pool: MAX
    kernel_size: 2
    kernel_depth: 2
    stride: 2
    temporal_stride: 2
  }
}
# ---------------- fc layers -------------
layers {
  name: "fc6"
  type: INNER_PRODUCT
  bottom: "pool5"
  top: "fc6"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  inner_product_param {
    num_output: 2048
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu6"
  type: RELU
  bottom: "fc6"
  top: "fc6"
}
layers {
  name: "drop6"
  type: DROPOUT
  bottom: "fc6"
  top: "fc6"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layers {
  name: "fc7"
  type: INNER_PRODUCT
  bottom: "fc6"
  top: "fc7"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  inner_product_param {
    num_output: 2048
    weight_filler {
      type: "gaussian"
      std: 0.005
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "relu7"
  type: RELU
  bottom: "fc7"
  top: "fc7"
}
layers {
  name: "drop7"
  type: DROPOUT
  bottom: "fc7"
  top: "fc7"
  dropout_param {
    dropout_ratio: 0.5
  }
}
layers {
  name: "fc8"
  type: INNER_PRODUCT
  bottom: "fc7"
  top: "fc8"
  blobs_lr: 1
  blobs_lr: 0
  weight_decay: 1
  weight_decay: 0
  inner_product_param {
    num_output: 3
    weight_filler {
      type: "gaussian"
      std: 0.01
    }
    bias_filler {
      type: "constant"
      value: 0
    }
  }
}
layers {
  name: "prob"
  type: SOFTMAX
  bottom: "fc8"
  top: "prob"
}
layers {
  top: "accuracy"
  name: "accuracy"
  type: ACCURACY
  bottom: "prob"
  bottom: "label"
}

**Comment by MarcoSaku** [comment]
I solved! I don't know why but if I paste the command instead to run the script it works. Maybe something related to file-format. Thanks for the support!

**Comment by dutran** [comment]
Glad to hear.

**Comment by wandering007** [comment]
@MarcoSaku faced with the same problem, don't know why either...

**Comment by xiaoxinlong** [comment]
> I solved! I don't know why but if I paste the command instead to run the script it works. Maybe something related to file-format. Thanks for the support!

thy to use 'dos2unix' to change the format of the shell file, such as "dos2unix -n c3d_sport1m_feature_extraction_frm.sh tt1.sh", then "sh tt1.sh",It's OK.
